### PR TITLE
[Snippets] Fixed output tensor names for wrap_as_subgraph

### PR DESCRIPTION
**Pull request by a-sidorova**
### Details:
 - *When we create subgraph from 1 node using `wrap_as_subgraph` method, we don't copy output tensor name and names from original node to body node. This PR fixes this problem. It's important when this alone node is before `Result` and he has names for output tensor to get blob data using tensor name*

### Tickets:
 - *N/A*

### CODE REVIEW

**Comment by a-sidorova** [comment]
@dmitry-gorokhov take a look please

File: src/common/snippets/src/op/subgraph.cpp [hunk 1]
```diff
@@ -92,6 +94,16 @@ auto snippets::op::Subgraph::wrap_node_as_subgraph(const std::shared_ptr<ov::Nod
...
+    NGRAPH_SUPPRESS_DEPRECATED_START
+    for (size_t i = 0; i < node->get_output_size(); i++) {
+        const std::string new_name = ngraph::op::util::get_ie_output_name(node);
+        auto& out_tensor = body_node->get_output_tensor(i);
+        out_tensor.set_name(new_name);
+        if (!node->get_output_tensor(i).get_names().empty()) {
+            out_tensor.set_names(node->get_output_tensor(i).get_names());
+        }
+    }
+    NGRAPH_SUPPRESS_DEPRECATED_END
```
Comment (dmitry-gorokhov) [inline src/common/snippets/src/op/subgraph.cpp:106]: As I can see we already have more or less the same logic inside `update_out_tensor_name`. Can we reuse it? Or maybe update `update_out_tensor_name` to handle internal body as well?

**Comment by dmitry-gorokhov** [comment]
Is it possible to cover the issue by test (snippets unit or functional)?

File: src/common/snippets/src/op/subgraph.cpp (hunk 1 above)
Comment (a-sidorova) [reply src/common/snippets/src/op/subgraph.cpp]: As we discuss offline, I have added separate function for copying output tensor names
Also I have added functional tests to cover several results + output tensor names

**Comment by dmitry-gorokhov** [comment]
@IvanNovoselov I think we need your review as well

**Comment by a-sidorova** [comment]
The changes were merged with the [PR](https://github.com/openvinotoolkit/openvino/pull/12384)

**Pull request closed without merging**